src/gwt/acesupport/snippets/snippets/markdown.js

```diff
@@ -0,0 +1,45 @@
+/*
+ * markdown.js
+ *
+ * Copyright (C) 2009-15 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+define("rstudio/snippets/markdown", function(require, exports, module) {
+
+var utils = require("rstudio/snippets");
+var SnippetManager = require("ace/snippets").snippetManager;
+
+var snippets = [
+   {
+      name: "[",
+      content: '[${1:label}](${2:location})'
+   },
+   {
+      name: "![",
+      content: '![${1:label}](${2:location})'
+   },
+   {
+      name: "r",
+      content: "```{r ${1:label}, ${2:options}}\n${0}\n```"
+   },
+   {
+      name: "rcpp",
```

Comment:
Note that the {Rcpp} syntax isn't in knitr yet so we should probably use the old engine= syntax for now